Ticker

Eric89GXL opened the pull request:
Good to go from my end. You can choose between a dumb one `linspace`, matplotlib's algorithm, and [Talbot's algorithm] (http://www.justintalbot.com/research/axis-labeling/comment-page-1/#comment-90585) by uncommenting lines. At some point we should make the API for this easy. I know you told me to subclass `Ticker`, but I think we might want to make it simpler. Users could provide a function that takes `(vmin, vmax)` and returns the ticks, as shown by the fact that you can just uncomment any of those three lines and be good to go. We should figure that out after scipy, though -- for now this should get you going.

I did notice a couple of odd behaviors.

1. The `n_inches` I compute above does not change when I resize my window. I must be using the transform system wrong, or the transform system is broken, but I assumed going from visual coords to document coords would get me the number of logical pixels, which I could then map using DPI into inches.

2. When I resize the window horizontally, the labeling on the axis changes even though the data does not. So there is some bug in the linking.

Any ideas about those?

Eric89GXL commented:
@campagnola I have confirmed that the weird behaviors noted above both occur on the current `scenegraph-update` branch, so they are not due to my changes. Assuming those are the behaviors you meant when you said it wasn't working properly, this is probably good to go.

campagnola commented:
What I mean is that when I zoom the view in `axes_plot.py`, the ticks no longer cover the entire length of the axis. It seems like this should not be possible ?

Eric89GXL commented:
How are you zooming? For some reason scroll wheel isn't doing it for me, if
it's supposed to.

campagnola commented:
right-drag. I'll fix the wheel issue.

Eric89GXL commented:
For some reason that doesn't work for me either :( Do you have this issue on my branch, but not on `scenegraph-update`?

Eric89GXL commented:
(I assume it's okay for me to rebase this to remove conflicts?)

Eric89GXL commented:
I think the incorrect length might be related to the fact that `n_inches` does not change as the canvas gets resized, but I'm not 100% sure. It might be hitting some degenerative case. For now I'll switch it to use the `matplotlib` ticking, which does not have this problem.

Eric89GXL commented:
Rebased, go ahead and try again. I can't zoom, so hopefully it works okay for you...

Eric89GXL commented:
Actually I think I fixed the bug with the Talbot ticking, so I switched it back :) Hopefully it works now. If it does not, you can try just commenting out the Talbot bit and using the `MaxNLocator` line, which is the matplotlib algorithm.

Eric89GXL commented:
This one should be good to go @campagnola, I think the problems it has now are also present in the current `scenegraph-update` branch

campagnola commented:
Thanks!

campagnola commented:
As an example of the problem I am having, you can try changing `examples/basics/visuals/line_plot_axes.py` such that the x-axis domain is (-1, 1):

```
        self.axis_x = visuals.AxisVisual(pos_xax, (-1, 1), (0., 1.))
```

Any idea how I can fix this?

Eric89GXL commented:
I'll try to take a look tonight or tomorrow. In the meantime, you could try just using the matplotlib line and see if that works...?

campagnola commented:
MPL ticker bombs out on some NaNs. I should be able to work on it tonight if you can't.

Eric89GXL commented:
Weird, that one was mostly copy-paste. Go ahead and work on it if you have time (I am unlikely to tonight), just ping me if you start/stop if possible so we don't overlap.

campagnola commented:
Ok, officially clocking in  :)

campagnola commented:
Easy!
https://github.com/campagnola/vispy/commit/4b4cb31f4409f130486151572d62b8602be08a11

Eric89GXL commented:
Oh, right -- my bad :)
